pyproject.toml

```diff
@@ -305,3 +305,14 @@ enable = [
     "deprecated-pragma",
     "invalid-name"
 ]
+
+[tool.bandit]
+exclude_dirs = [
+    "test",
+    ".eggs"
+]
+skips = [
+    "B101", # assert_used
```

Comment:
We use `assert` pervasively in order to convince the typechecker that an `Optional` is not None. We should evaluate all these Optionals and see if they should ever be None.

I didn't want to pollute the code with `if None... raise` checks all over the place.